### PR TITLE
Add Repair only broken Weapon and Consume on Use RepairKit

### PR DESCRIPTION
**Pull request by thierryboiago**
- added setting to use repair only when weapon is broken.
- has been added so when repair kit is used it disappears regardless of durability

### CODE REVIEW

**Comment by thierryboiago** [comment]
image

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java [hunk 1]
```diff
@@ -136,15 +136,22 @@ public void onInventoryClick(InventoryClickEvent event) {
             // complicated, so we handle it separately.
             if (CustomTag.BROKEN_WEAPON.hasString(weapon)) {
                 repairBrokenItem(event);
-                return;
-            }
 
-            // Only attempt to repair guns with proper repair items
-            if (weaponTitle == null || event.getCursor() == null)
-                return;
+            }else{
```
Comment (CJCrafter) [inline WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java:140]: We use oracle conventions in the WeaponMechanics repo `}else{` -> `} else {`

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java [hunk 2]
```diff
@@ -223,6 +230,10 @@ public boolean repair(ItemStack weapon, String weaponTitle, ItemStack repairItem
                 if (kit.getBreakMechanics() != null) kit.getBreakMechanics().use(cast);
             }
 
+            if(kit.consumeOnUse){
```
Comment (CJCrafter) [inline WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java:233]: if (kit.consumeOnUse) {

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java [hunk 3]
```diff
@@ -316,22 +327,23 @@ public static class RepairKit implements Serializer<RepairKit> {
         private Set<String> weapons;
         private Set<String> armors;
         private Mechanics breakMechanics;
-
+        private boolean consumeOnUse;
```
Comment (CJCrafter) [inline WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java:330]: blank line before javadoc comments

File: WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java [hunk 4]
```diff
@@ -136,15 +136,22 @@ public void onInventoryClick(InventoryClickEvent event) {
...
+
+                // Only attempt to repair guns with proper repair items
+                if (weaponTitle == null || event.getCursor() == null)
+                    return;
 
-            CastData cast = new CastData(event.getWhoClicked(), weaponTitle, weapon);
-            repair(weapon, weaponTitle, event.getCursor(), cast);
+                Configuration config = WeaponMechanics.getConfigurations();
+                CustomDurability customDurability = config.getObject(weaponTitle + ".Shoot.Custom_Durability", CustomDurability.class);
+                if (customDurability.isRepairOnlyBroken()){
```
Comment (CJCrafter) [inline WeaponMechanics/src/main/java/me/deecaad/weaponmechanics/listeners/RepairItemListener.java:148]: if (customDurability.isRepairOnlyBroken()) {

**Review by CJCrafter — CHANGES_REQUESTED** [review]

**Review by CJCrafter — APPROVED** [review]
Looks good!

**Pull request merged**